### PR TITLE
Session iOS

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* add sessions on iOS
* minor example app UI improvements

### CODE REVIEW

File: ios/Components/SessionHelperModule.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        } catch {
+            return rejecter("session", nil, error)
+        }
+
+        guard let id = sessionModelJSON["id"] as? String, let data = sessionModelJSON["sessionData"] as? String else {
+            return rejecter("session", "Invalid session data", nil)
+        }
+
+        let config = AdyenSession.Configuration(sessionIdentifier: id, initialSessionData: data, context: context)
+        DispatchQueue.main.async {
```
Comment (goergisn) [inline ios/Components/SessionHelperModule.swift:51]: Is this super necessary?

File: ios/Components/SessionHelperModule.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                }
+            }
+        }
+    }
+
+    @objc
+    func hide(_ success: NSNumber, event: NSDictionary) {
+        dismiss(success.boolValue)
+    }
+
```
Comment (goergisn) [inline ios/Components/SessionHelperModule.swift:69]: Maybe 1 or 2 less newlines

File: ios/Components/SessionHelperModule.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    BaseModule.session = session
+                case let .failure(error):
+                    rejecter("session", nil, error)
+                }
+            }
+        }
+    }
+
+    @objc
+    func hide(_ success: NSNumber, event: NSDictionary) {
```
Comment (goergisn) [inline ios/Components/SessionHelperModule.swift:66]: Do we need to use NextStep objects here?

**Review by goergisn — DISMISSED** [review]

File: ios/Components/SessionHelperModule.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    func didFail(with error: Error, from component: Adyen.Component, session: Adyen.AdyenSession) {
+        SessionHelperModule.sessionListener?.didFail(with: error)
+    }
+
+    func didOpenExternalApplication(component: Adyen.ActionComponent, session: Adyen.AdyenSession) {
+    }
+
+    override public func supportedEvents() -> [String]! { [Events.didComplete.rawValue, Events.didFail.rawValue] }
+
```
Comment (Robert-SD) [inline ios/Components/SessionHelperModule.swift:32]:
```suggestion
```

File: ios/Components/SessionHelperModule.swift (hunk 1 above)
Comment (descorp) [reply ios/Components/SessionHelperModule.swift:51]: yep 🤷‍♂️
React Native calling native modules in non-UI thread

File: ios/Components/SessionHelperModule.swift (hunk 2 above)
Comment (descorp) [reply ios/Components/SessionHelperModule.swift:69]: Thanks :)
Fixed in [f53fb0f](https://github.com/Adyen/adyen-react-native/pull/332/commits/f53fb0fe93e637ff8bc493ac6277ad1d583a3779)

File: ios/Components/SessionHelperModule.swift (hunk 3 above)
Comment (descorp) [reply ios/Components/SessionHelperModule.swift:66]: yep 🤷‍♂️
React Native is C++ based and operating via [obj-c primitives](https://reactnative.dev/docs/native-modules-ios#argument-types)

**Review by goergisn — APPROVED** [review]

**Pull request merged**